4-fleetscope/modules/env_baseline/main.tf

```diff
@@ -14,4 +14,34 @@
  * limitations under the License.
  */
 
-#TODO Add GKE cluster definition
+locals {
+  membership_re = "/memberships/([^/]*)$"
+}
+
+resource "random_string" "suffix" {
+  length  = 6
+  special = false
+  upper   = false
+}
+
+resource "google_gke_hub_scope" "fleet-scope" {
+  scope_id = "${var.scope_id}-${var.env}"
+  project  = var.project_id
+}
+
+resource "google_gke_hub_namespace" "fleet-ns" {
+  scope_namespace_id = "${var.namespace_id}-${var.env}"
+  scope_id           = google_gke_hub_scope.fleet-scope.scope_id
+  scope              = google_gke_hub_scope.fleet-scope.name
+  project            = var.project_id
+}
+
+resource "google_gke_hub_membership_binding" "membership-binding" {
+  for_each = toset(var.cluster_membership_ids)
+
+  membership_binding_id = "${var.scope_id}-${var.env}-${random_string.suffix.result}"
+  scope                 = google_gke_hub_scope.fleet-scope.name
+  membership_id         = regex(local.membership_re, each.key)[0]
+  location              = "global"
```

Comment:
FYI: In my next update to (https://github.com/GoogleCloudPlatform/terraform-google-enterprise-application/blob/main/2-multitenant/modules/env_baseline/main.tf), the memberships will be properly regionalized. I'd suggest parsing out similar to the `membership_id`, or at least add a TODO.

Here is what I've used before (also pulls the cluster's project): `membership_re   = "//gkehub.googleapis.com/projects/([^/]*)/locations/([^/]*)/memberships/([^/]*)$"`

Reply:
sounds good, thanks for suggestion.